### ISSUE TITLE
max_connections circuit breaker not always honored when using multiple priorities

### ISSUE THREAD

**Issue by snowp** [envoyproxy/envoy#6476]
When a stream is created, the http1 connection pool first attempts to use an existing connection. If one is not available, then the max_pending and max_connections circuit breaker is consulted before creating a new connection.

When using multiple priorities with different values for max_connections this can let the lower value exceed its circuit breaker:

DEFAULT has max 10
HIGH has max 100

Envoy gets a bunch of requests at once, depletes the 10 for the DEFAULT requests, but allow the HIGH requests to make new connections. At this point there are >10 available connections, allowing a higher than desired concurrency for DEFAULT requests. 

Note that this is mostly theoretical based on me reading the code, I could totally be wrong here.

**Comment by snowp** [comment]
I came over this trying to figure out why setting max_connections to 0 doesn't seem to do what I'd expect (fail all traffic). Guessing there's somehow a connection being created somewhere before checking the resource count, and since we don't look at it before assigning a request to an existing connection it will happily assign the request to the open connection.

**Comment by mattklein123** [comment]
@snowp IIRC the different priorities use different resource managers, so I don't think they interact at all? Can you point me at the code you think is doing this?

**Comment by snowp** [comment]
My bad - I missed the part that there is one connection pool per priority. My thinking was that there was a shared pool, so that connections could be created through quotas in one priority and then used in the other. 

tl;dr I didn't realize `priority_` in `conn_pool.cc` was a member variable. :) Sorry for the noise
